oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/MultithreadedTraverseWithSortStrategy.java

```diff
@@ -304,14 +364,18 @@ void createInitialTasks(NodeStateEntryTraverserFactory nodeStateEntryTraverserFa
     void addTask(TraversingRange range, NodeStateEntryTraverserFactory nodeStateEntryTraverserFactory, BlobStore blobStore,
                          ConcurrentLinkedQueue<String> completedTasks) throws IOException {
         taskQueue.add(new TraverseAndSortTask(range, comparator, blobStore, storeDir,
-                compressionEnabled, completedTasks, taskQueue, phaser, nodeStateEntryTraverserFactory, memoryManager, dumpThreshold));
+                compressionEnabled, completedTasks, taskQueue, phaser, nodeStateEntryTraverserFactory, memoryManager, dumpThreshold, sortedFiles));
     }
 
     @Override
     public File createSortedStoreFile() throws IOException, CompositeException {
         String watcherThreadName = "watcher";
+        String mergerThreadName = "merger";
         Thread watcher = new Thread(new TaskRunner(), watcherThreadName);
         watcher.start();
+        File sortedFile = new File(storeDir, getSortedStoreFileName(compressionEnabled));
+        Thread merger = new Thread(new MergeRunner(sortedFile), mergerThreadName);
```

Comment:
Could you set setDaemon(true) please? So that the JVM will stop if this is the only living thread. I see you have already set the thread name, that's good.

Reply:
make sense